### PR TITLE
Refactor ios play input sound logic.

### PR DESCRIPTION
**Pull request by kangwang1988**
As there are extra requirement to use playInputClick in iOS, use AudioServicesPlaySystemSound instead to play input sound.
This will also fix https://github.com/flutter/flutter/issues/27768 .
See http://iphonedevwiki.net/index.php/AudioServices for the sound codes.

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm [hunk 1]
```diff
@@ -93,9 +93,8 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
 - (void)playSystemSound:(NSString*)soundType {
   if ([soundType isEqualToString:@"SystemSoundType.click"]) {
     // All feedback types are specific to Android and are treated as equal on
-    // iOS. The surface must (and does) adopt the UIInputViewAudioFeedback
-    // protocol
-    [[UIDevice currentDevice] playInputClick];
+    // iOS.
+    AudioServicesPlaySystemSound(1306);
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm]: Please extract this to a named constant; e.g. kKeypressClickSound.

**Comment by kangwang1988** [comment]
@cbracken 
Done.

**Comment by CharlieScarver** [comment]
@cbracken @chinmaygarde 
Any updates on this? Is it going to be merged soon?

**Comment by cbracken** [comment]
Sorry for the delay, this slipped off my radar. Thanks for your contribution!

**Pull request merged**